model/src/main/proto/oppia_logger.proto

```diff
@@ -110,159 +130,156 @@ message EventLog {
-    ExplorationData exploration_data= 2;
-  }
-
-  // Structure of a submit answer context.
+  // Represents the event context for submitting an answer within an exploration.
   message SubmitAnswerContext {
-    GenericData generic_data = 1;
-    ExplorationData exploration_data= 2;
+    // Defined attributes that are common among other exploration related event log contexts.
+    ExplorationDetailsContext exploration_details = 1;
+
     // Whether the answer that the learner submitted is the correct answer.
     bool is_answer_correct = 3;
   }
 
-  // Structure of a play voiceover context.
+  // Represents the event context for playing a voiceover within an exploration.
   message PlayVoiceOverContext {
-    GenericData generic_data = 1;
-    ExplorationData exploration_data= 2;
-    // The content id of the voiceover recording.
+    // Defined attributes that are common among other exploration related event log contexts.
+    ExplorationDetailsContext exploration_details = 1;
+
+    // The content id of the voiceover recording being played.
     string content_id = 3;
   }
 
-  // Structure of an app in background context.
-  message AppInBackgroundContext {
-    GenericData generic_data = 1;
-  }
+  // Represents event context which contains learner-specific details that are logged with every event.
+  message LearnerDetailsContext {
+    // The learner id of the current user. This id is profile-specific.
+    string learner_id = 1;
 
-  // Structure of an app in foreground context.
-  message AppInForegroundContext {
-    GenericData generic_data = 1;
+    // The device id of the current device. This id is common across all profiles.
+    string device_id = 2;
   }
 
-  // Structure of an exit exploration context.
-  message ExitExplorationContext {
-    GenericData generic_data = 1;
-    ExplorationData exploration_data= 2;
-  }
+  // Represents the event context for all exploration related events containing learner-specific and
+  // exploration-specific details.
+  message ExplorationDetailsContext {
+    // The session id of the current learning session.
+    string session_id = 1;
 
-  // Structure of a finish exploration context.
-  message FinishExplorationContext {
-    GenericData generic_data = 1;
-    ExplorationData exploration_data= 2;
-  }
+    // The ID of the exploration.
+    string exploration_id = 2;
 
-  // Structure of a resume exploration context.
-  message ResumeExplorationContext {
-    GenericData generic_data = 1;
-  }
+    // The version of the exploration.
+    string exploration_version = 3;
 
-  // Structure of a start over exploration context.
-  message StartOverExplorationContext {
-    GenericData generic_data = 1;
-  }
+    // The name of the current state of the exploration.
+    string state_name = 4;
 
-  // Structure of a delete profile context.
-  message DeleteProfileContext {
-    GenericData generic_data = 1;
+    // Defined attributes that are common among all event log contexts.
```

Comment:
Not all--please be more specific here (since we already log events that don't include these details).

Reply:
done.